test/functional/test_framework/test_node.py

```diff
@@ -97,14 +97,16 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             "-debugexclude=leveldb",
             "-uacomment=testnode%d" % i,
         ]
-        if use_valgrind:
+        if self.is_valgrind_present() and use_valgrind:
+            exit_on_first_error = self.should_exit_valgrind_on_first_error()
             default_suppressions_file = os.path.join(
                 os.path.dirname(os.path.realpath(__file__)),
                 "..", "..", "..", "contrib", "valgrind.supp")
             suppressions_file = os.getenv("VALGRIND_SUPPRESSIONS_FILE",
                                           default_suppressions_file)
             self.args = ["valgrind", "--suppressions={}".format(suppressions_file),
-                         "--gen-suppressions=all", "--exit-on-first-error=yes",
+                         "--gen-suppressions=all",
+                         "--exit-on-first-error=yes" if exit_on_first_error else "--exit-on-first-error=no",
```

Comment:
perhaps construct this string outside of the arg array assignment, or do:
```python3
 "--exit-on-first-error={}".format("yes" if exit_on_first_error else "no"),
```

Reply:
Many thanks! I am going through the comments and will adapt the code accordingly.

Reply:
`--exit-on-first-error=yes` should always be used if the option `--exit-on-first-error` is supported :)

Reply:
Would prefer your solution, but because not all old versions support this flag I'll have to stick with the other variant. Will basically return an empty string when the flag should not be active, because this is the default for some older versions anyway.

Reply:
> `--exit-on-first-error=yes` should always be used if the option `--exit-on-first-error` is supported :)

But the original discussion was about disabling it for v3.15. Or am I missing something?

Reply:
I'd rename `exit_on_first_error` to `supports_ exit_on_first_error` for clarity.

Reply:
Done!

Reply:
> I'd rename `exit_on_first_error` to `supports_exit_on_first_error` for clarity.

Then I will also put the expected version (major, minor) to 3.14, ok?
If I understand @practicalswift correctly, every version that supports this flag should also have it activated by default.

Reply:
> If I understand @practicalswift correctly, every version that supports this flag should also have it activated by default.

Correct! :)